python/triton/language/semantic.py

```diff
@@ -1303,7 +1303,7 @@ def atom_red_typechecking_impl(ptr: tl.tensor, val: tl.tensor, mask: tl.tensor,
     element_ty = ptr.type.scalar.element_ty
     if element_ty is tl.float16 and op != 'add':
         raise ValueError("atomic_" + op + " does not support fp16")
-    if element_ty in [tl.int1, tl.int8, tl.int16, tl.bfloat16]:
+    if element_ty in [tl.int1, tl.int8, tl.int16]:
```

Comment:
Actually one more thing: should we also change L1304 to only allow atomic_add with bf16? I don't think other atomic ops support bf16 yet?